plugins/field-colour/src/blocks/colourBlend.ts

```diff
@@ -0,0 +1,296 @@
+/**
+ * @license
+ * Copyright 2024 Google LLC
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import {Block, common as BlocklyCommon} from 'blockly';
+import {
+  JavascriptGenerator,
+  Order as JavascriptOrder,
+} from 'blockly/javascript';
+import {DartGenerator, Order as DartOrder} from 'blockly/dart';
+import {LuaGenerator, Order as LuaOrder} from 'blockly/lua';
+import {PhpGenerator, Order as PhpOrder} from 'blockly/php';
+import {PythonGenerator, Order as PythonOrder} from 'blockly/python';
+import {registerFieldColour} from '../field_colour';
+import {Generators} from './generatorUtils';
```

Comment:
Per practice in the core repo, I think imports of files from the same package should use the full (compiled) filename:

```suggestion
import {registerFieldColour} from '../field_colour.js';
import {Generators} from './generatorUtils.js';
```

This doesn't matter so long as we are webpacking the file, but will matter if we start publishing unbundled packages (which we are intending to).

(Here and most files below.0

Reply:
Changing this to use `.js` extensions failed with this error:
```
ERROR in ./src/blocks/colourBlend.ts 12:0-57
Module not found: Error: Can't resolve '../field_colour.js' in '/home/fenichel/Documents/blockly-dev/blockly-samples/plugins/field-colour/src/blocks'
resolve '../field_colour.js' in '/home/fenichel/Documents/blockly-dev/blockly-samples/plugins/field-colour/src/blocks'
  using description file: /home/fenichel/Documents/blockly-dev/blockly-samples/plugins/field-colour/package.json (relative path: ./src/blocks)
    Field 'browser' doesn't contain a valid alias configuration
    using description file: /home/fenichel/Documents/blockly-dev/blockly-samples/plugins/field-colour/package.json (relative path: ./src/field_colour.js)
      no extension
        Field 'browser' doesn't contain a valid alias configuration
        /home/fenichel/Documents/blockly-dev/blockly-samples/plugins/field-colour/src/field_colour.js doesn't exist
      .ts
        Field 'browser' doesn't contain a valid alias configuration
        /home/fenichel/Documents/blockly-dev/blockly-samples/plugins/field-colour/src/field_colour.js.ts doesn't exist
      .js
        Field 'browser' doesn't contain a valid alias configuration
        /home/fenichel/Documents/blockly-dev/blockly-samples/plugins/field-colour/src/field_colour.js.js doesn't exist
      as directory
        /home/fenichel/Documents/blockly-dev/blockly-samples/plugins/field-colour/src/field_colour.js doesn't exist
 @ ./src/index.ts 10:0-52 17:0-54 18:0-40 30:4-28
 @ ./test/index.ts 16:0-71 272:4-23
 ```